libcontainer/utils/utils_unix.go

```diff
@@ -23,9 +26,44 @@ func EnsureProcHandle(fh *os.File) error {
 	return nil
 }
 
+var (
+	haveCloseRangeCloexec     bool
+	haveCloseRangeCloexecOnce sync.Once
+)
+
+func tryCloseRangeCloexec(minFd int) error {
+	haveCloseRangeCloexecOnce.Do(func() {
+		// Make sure we're not closing a random file descriptor.
+		tmpFd, err := unix.FcntlInt(0, unix.F_DUPFD_CLOEXEC, 0)
+		if err != nil {
+			return
+		}
+		defer unix.Close(tmpFd)
+
+		err = unix.CloseRange(uint(tmpFd), uint(tmpFd), unix.CLOSE_RANGE_CLOEXEC)
+		// Any error means we cannot use close_range(CLOSE_RANGE_CLOEXEC).
+		// -ENOSYS and -EINVAL ultimately mean we don't have support, but any
+		// other potential error would imply that even the most basic close
+		// operation wouldn't work.
+		haveCloseRangeCloexec = err == nil
+	})
+
+	var err error
+	if haveCloseRangeCloexec {
+		err = unix.CloseRange(uint(minFd), math.MaxUint, unix.CLOSE_RANGE_CLOEXEC)
+	} else {
+		err = unix.ENOSYS
+	}
+	return err
+}
+
 // CloseExecFrom applies O_CLOEXEC to all file descriptors currently open for
 // the process (except for those below the given fd value).
 func CloseExecFrom(minFd int) error {
+	if err := tryCloseRangeCloexec(minFd); !errors.Is(err, unix.ENOSYS) {
+		return err
```

Comment:
Here `err` is a bare error (a mere `errno`), meaning two things:
1. We can use `err != unix.ENOSYS` for comparison which is somewhat faster (but requires `//nolint:errorlint // unix errors are bare` to silence a linter, see https://github.com/polyfloyd/go-errorlint/issues/10);
2. It's better to wrap the error before returning it, e.g. `return &os.SyscallError{Syscall: "close_range", Err: err}`.

Reply:
I always forget to do this for `unix` errors, sorry. I'll do the latter.